licence-checker/licence-checker.py

```diff
@@ -157,23 +165,24 @@ def full_line_parts(self, licence_line):
 
 class LicenceMatcher:
     '''An object to match a given licence at the start of a file'''
-    def __init__(self, comment_style, licence):
+    def __init__(self, comment_style, licence, match_regex):
         self.style = comment_style
 
-        self.search_marker = comment_style.search_line(licence.first_word)
+        self.search_marker = comment_style.search_line(licence.first_word,
+                                                       match_regex)
```

Comment:
I think this doesn't make sense if `match_regex` is true. I'm not sure what the `first_word` tracking tries to catch (@lenary: can you comment?), but I think the only sensible thing you can do when `match_regex` is true is use a "first_line" instead.